src/mon/OSDMonitor.cc

```diff
@@ -989,11 +988,37 @@ int OSDMonitor::dump_osd_metadata(int osd, Formatter *f, ostream *err)
       *err << "osd." << osd << " metadata is corrupt";
     return -EIO;
   }
+  return 0;
+}
+
+int OSDMonitor::dump_osd_metadata(int osd, Formatter *f, ostream *err)
+{
+  map<string,string> m;
+  if (int r = load_metadata(osd, m, err))
+    return r;
   for (map<string,string>::iterator p = m.begin(); p != m.end(); ++p)
     f->dump_string(p->first.c_str(), p->second);
   return 0;
 }
 
+void OSDMonitor::print_nodes(Formatter *f)
+{
+  f->open_object_section("osd");
+  for (int osd = 0; osd <= osdmap.get_max_osd(); osd++) {
```

Comment:
the group-by-host feature sounds great. but in a large cluster, to group the OSD hosts could be expensive, i guess.